spec/services/fees/charge_service_spec.rb

```diff
@@ -42,170 +42,911 @@
+
+        aggregate_failures do
+          expect(created_fees.count).to eq(3)
+          expect(created_fees).to all(
+            have_attributes(
+              invoice_id: invoice.id,
+              charge_id: charge.id,
+              amount_currency: 'EUR',
+              vat_rate: 20.0,
+            ),
+          )
+          expect(created_fees.first).to have_attributes(
+            amount_cents: 4000,
+            vat_amount_cents: 800,
+            units: 2,
+          )
+          expect(created_fees.second).to have_attributes(
+            amount_cents: 5000,
+            vat_amount_cents: 1000,
+            units: 1,
+          )
+          expect(created_fees.third).to have_attributes(
+            amount_cents: 4000,
+            vat_amount_cents: 800,
+            units: 1,
+          )
+        end
+      end
+
+      it 'creates expected fees for recurring_count_agg aggregation type' do
+        boundaries = {
+          from_date: subscription.started_at.at_beginning_of_month.next_month.to_date,
+          to_date: subscription.started_at.next_month.end_of_month.to_date,
+          charges_from_date: subscription.started_at.at_beginning_of_month.next_month.to_date,
+          charges_to_date: subscription.started_at.next_month.end_of_month.to_date,
+        }
+
+        create(
+          :persisted_event,
+          customer: subscription.customer,
+          billable_metric: billable_metric,
+          external_subscription_id: subscription.external_id,
+          external_id: 'ext_11',
+          added_at: subscription.started_at - 1.day,
+          properties: {
+            'operation_type' => 'add',
+            'unique_id' => 'ext_123',
+            'region' => 'usa',
+            'foo_bar' => 12,
+          },
+        )
+        create(
+          :persisted_event,
+          customer: subscription.customer,
+          billable_metric: billable_metric,
+          external_subscription_id: subscription.external_id,
+          external_id: 'ext_12',
+          added_at: subscription.started_at - 1.day,
+          properties: {
+            'operation_type' => 'add',
+            'unique_id' => 'ext_456',
+            'region' => 'europe',
+            'foo_bar' => 10,
+          },
+        )
+        create(
+          :persisted_event,
+          customer: subscription.customer,
+          billable_metric: billable_metric,
+          external_subscription_id: subscription.external_id,
+          external_id: 'ext_13',
+          added_at: subscription.started_at - 1.day,
+          properties: {
+            'operation_type' => 'add',
+            'unique_id' => 'ext_789',
+            'country' => 'france',
+            'foo_bar' => 5,
+          },
+        )
+
+        billable_metric.update!(aggregation_type: :recurring_count_agg, field_name: 'foo_bar')
+        result = described_class.new(invoice: invoice, charge: charge, subscription: subscription, boundaries: boundaries).create
```

Comment:
Layout/LineLength: Line is too long. [129/120] (https://rubystyle.guide#max-line-length)